### PR TITLE
Modify custom test w/ regard to timing and expected values

### PR DESCRIPTION
**Pull request by rafsalas19**
- Parallelize ib write test
- Modify a few other test so that they do not return a 0 value. Needed for another workflow
- bug fixes
- increase time out due to tests taking a bit longer than 6 minutes
- All changes tested at scale

### CODE REVIEW

File: customTests/azure_cuda_bandwidth.nhc [hunk 1]
```diff
@@ -50,15 +50,15 @@ function check_cuda_bw()
 	lock_clocks="sudo nvidia-smi -lgc 1400"
 	result=$(catch_error "$lock_clocks" "$error_smi")
     if ! catch_error "$lock_clocks" "$error_smi"; then
```
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc]: Is something weird going on with the indenting here?

File: customTests/azure_cuda_bandwidth.nhc [hunk 2]
```diff
@@ -67,7 +67,7 @@ function check_cuda_bw()
 	error_htod="**Fail** The htod gpu_copy test failed to execute."
 	error_htod+="It exited with error code"
 	if ! catch_error "$exec_htod" "$error_htod"; then
-        return 1
+        return 0
```
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc]: Again what's going on with the indenting here?

File: customTests/azure_cuda_bandwidth.nhc
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc:1]: does "declare -g output=$($1)" actually capture the return values of the executable $1? Testing it out it seems to always return 0?

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: let me verify. I didn't notice this breaking but let me check

File: customTests/azure_cuda_bandwidth.nhc (hunk 1 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: must be switching from vim to vscode let me take a look

File: customTests/azure_cuda_bandwidth.nhc (hunk 2 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: will look into

File: customTests/azure_cuda_bandwidth.nhc
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: So I just ran the script as a bash script and echo'd the contents both in the check function and the caller function and both output the expected GPU BW values.

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: try this:
declare -g output=$(hostname)
echo $output

File: customTests/azure_cuda_bandwidth.nhc (hunk 1 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: addressed

File: customTests/azure_cuda_bandwidth.nhc (hunk 2 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: addressed

File: customTests/azure_cuda_bandwidth.nhc
Comment (moes1) [reply customTests/azure_cuda_bandwidth.nhc:1]: Sorry I should be more clear. What I was asking about is this line: 

https://github.com/Azure/azurehpc-health-checks/blob/4fb9d3b5a9e0bcfb522e29781cdffb6e80e805fc/customTests/azure_cuda_bandwidth.nhc#L14

I think the return code might always be zero even if the command run as "$($1)" in the previous line failed.

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: oh I see let me see if I can make this work

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: Ok I was able to address this by separating the declaration from the assignment of the output variable.

**Review by moes1 — APPROVED** [review]

**Pull request merged**